src/webview/tests.py

```diff
@@ -51,3 +57,117 @@ def mock_login(
         self.assertEqual(location, self.entry_point)
+
+
+class AddMaintainerViewTests(TestCase):
+    def setUp(self) -> None:
+        # Create user and log in
+        self.user = User.objects.create_user(username="admin", password="pw")
+        self.user.is_staff = True
+        self.user.save()
+        self.client = Client()
+        self.client.login(username="admin", password="pw")
+
+        # Create a maintainer
+        self.maintainer = NixMaintainer.objects.create(
+            github_id=123,
+            github="existinguser",
+            name="Existing User",
+            email="existing@example.com",
+        )
+
+        # Create a suggestion and cached suggestion with the maintainer
+        self.assigner = Organization.objects.create(uuid=1, short_name="foo")
+        self.cve_record = CveRecord.objects.create(
+            cve_id="CVE-2025-0001",
+            assigner=self.assigner,
+        )
+        self.suggestion = CVEDerivationClusterProposal.objects.create(
+            status=CVEDerivationClusterProposal.Status.PENDING,
+            cve_id=self.cve_record.id,
+        )
+        self.cached = CachedSuggestions.objects.create(
+            proposal=self.suggestion,
+            payload={
+                "maintainers": [
+                    {
+                        "github_id": 123,
+                        "github": "existinguser",
+                        "name": "Existing User",
+                        "email": "existing@example.com",
+                    }
+                ],
+                "packages": {},
+                "title": "Test Suggestion",
+            },
+        )
+
+    def test_add_existing_maintainer_returns_error(self) -> None:
+        url = reverse("webview:add_maintainer")
+        response = self.client.post(
+            url,
+            {
+                "suggestion_id": self.suggestion.pk,
+                "new_maintainer_github_handle": "existinguser",
+            },
+        )
+        self.assertEqual(response.context["error_msg"], "Already a maintainer")
+
+    def test_add_new_unknown_maintainer_success(self) -> None:
+        url = reverse("webview:add_maintainer")
+        # Simulate GitHub API returning a new user
+        with patch("webview.views.fetch_user_info") as mock_fetch:
+            mock_fetch.return_value = {
+                "id": 456,
+                "login": "newuser",
+                "name": "New User",
+                "email": "new@example.com",
+            }
+            response = self.client.post(
+                url,
+                {
+                    "suggestion_id": self.suggestion.pk,
+                    "new_maintainer_github_handle": "newuser",
+                },
+            )
+        self.assertEqual(response.status_code, 200)
+        # Reload cached suggestion and check maintainers
+        self.cached.refresh_from_db()
+        maintainers = self.cached.payload["maintainers"]
```

Comment:
Maybe we should also check that we're not failing at cache invalidation and compare with database contents.

Reply:
https://github.com/Nix-Security-WG/nix-security-tracker/pull/593/commits/f1871640907dcf42b87e78751c51cd5a3a71741d